frontend/src/searchBar/searchBar.component.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { Search } from '@mui/icons-material';
+import {
+  Avatar,
+  CircularProgress,
+  InputAdornment,
+  InputBase,
+  List,
+  ListItem,
+  ListItemText,
+  Popover,
+} from '@mui/material';
+import axios from 'axios';
+import debounce from 'lodash/debounce';
+
+import { Element } from '../interfaces/Element.component';
+import styles from './searchBar.module.css';
+
+interface SearchBarProps {
+  onSelect: (element: Element) => void;
+  isOpen: boolean;
+  setIsOpen: (open: boolean) => void;
+  searchTerm: string;
+  setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
```

Comment:
There's no need of importing whole React class from library. Import only things you need

import { useCallback, useEffect, useState, Dispatch, SetStateAction } from 'react';

```suggestion
  setSearchTerm: Dispatch<SetStateAction<string>>;
```